File: backend/app/schemas/note.py
from datetime import datetime

from pydantic import BaseModel


class SavedByUserSummary(BaseModel):
    id: str
    email: str
    first_name: str | None
    last_name: str | None


class NoteVersionResponse(BaseModel):
    id: str
    note_id: str
    version_number: int
    saved_by_user_id: str
    saved_by_user: SavedByUserSummary
    subjective: str | None
    objective: str | None
    assessment: str | None
    plan: str | None
    icd10_codes: list[dict[str, str]] | None
    saved_at: datetime
    generation_metadata: dict | None

File: frontend/types/note.ts
export interface SavedByUserSummary {
  id: string;
  email: string;
  first_name: string | null;
  last_name: string | null;
}

export interface NoteVersion {
  id: string;
  note_id: string;
  version_number: number;
  saved_by_user_id: string;
  saved_by_user: SavedByUserSummary;
  subjective: string | null;
  objective: string | null;
  assessment: string | null;
  plan: string | null;
  icd10_codes: Array<Record<string, string>> | null;
  saved_at: string;
  generation_metadata: Record<string, unknown> | null;
}
